p2p/protocol/holepunch/filter.go

```diff
@@ -0,0 +1,42 @@
+package holepunch
+
+import (
+	"github.com/libp2p/go-libp2p/core/peer"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+// WithAddrFilter is a Service option that enables multi address filtering.
+// It allows to only allow a subset of observed addresses to the remote
+// peer. E.g., only announce TCP or QUIC multi addresses instead of both.
+// It also allows to only consider a subset of received multi addresses
+// that remote peers announced to us.
+// Theoretically, this API also allows to add multi addresses in both cases.
+func WithAddrFilter(maf AddrFilter) Option {
+	return func(hps *Service) error {
+		hps.filter = maf
+		return nil
+	}
+}
+
+// AddrFilter defines the interface for the multi address filtering.
+//   - FilterLocal is a function that filters the multi addresses that
```

Comment:
Can you move these comments directly above the function definition in the interface? That way, they'll show up in IDEs.